core/services/pg/event_broadcaster.go

```diff
@@ -254,15 +254,15 @@ func (sub *subscription) Send(event Event) {
 const broadcastTimeout = 10 * time.Second
 
 func (sub *subscription) processQueue() {
-	ctx, cancel := context.WithTimeout(context.Background(), broadcastTimeout)
+	ctx, cancel := utils.ContextFromChanWithDeadline(sub.chDone, broadcastTimeout)
```

Comment:
This context actually looks pointless. How about using `time.After` instead?

Reply:
Sure, makes sense to me